src/visualization/makie_visualization.jl

```diff
@@ -0,0 +1,373 @@
+  Vplot1D = face_plotting_interpolation_matrix(dg; nvisnodes=nvisnodes)
+
+  # These 5 lines extract the face values on each element from the arrays x,y,sol_to_plot.
+  # The resulting arrays are then reshaped so that xf, yf, sol_f are Matrix types of size
+  # (Number of face plotting nodes) x (Number of faces).
+  function face_first_reshape(x, num_nodes_1D, num_nodes, num_elements)
+      num_reference_faces = 2 * ndims(mesh)
+      xf = view(reshape(x, num_nodes, num_elements), vec(Fmask), :)
+      return reshape(xf, num_nodes_1D, num_elements * num_reference_faces)
+  end
+  reshape_and_interpolate(x) = Vplot1D * face_first_reshape(x, num_nodes_1D, num_nodes, num_elements)
+  xfp, yfp = map(reshape_and_interpolate, (x, y))
+  ufp = StructArray{SVector{nvars, uEltype}}(map(reshape_and_interpolate, StructArrays.components(u)))
+
+  # convert variables based on solution_variables mapping
+  solution_variables_ = digest_solution_variables(equations, solution_variables)
+  variable_names = SVector(varnames(solution_variables_, equations))
+
+  transform_to_solution_variables!(uplot, solution_variables_, equations)
+  transform_to_solution_variables!(ufp, solution_variables_, equations)
+
+  return UnstructuredPlotData2D(xplot, yplot, uplot, t, xfp, yfp, ufp, variable_names)
+end
+
+function generate_plotting_triangulation(pds::PlotDataSeries2D{<:UnstructuredPlotData2D};
+                                         set_z_coordinate_zero = false)
+
+  @unpack variable_id = pds
+  pd = pds.plot_data
+  @unpack x, y, u, t = pd
+
+  makie_triangles = Makie.to_triangles(t)
+
+  # trimesh[i] holds GeometryBasics.Mesh containing plotting information on the ith element.
+  num_plotting_nodes, num_elements = size(x)
+  trimesh = Vector{GeometryBasics.Mesh{3, Float32}}(undef, num_elements)
+  coordinates = zeros(Float32, num_plotting_nodes, 3)
+  for element in Base.OneTo(num_elements)
+    for i in Base.OneTo(num_plotting_nodes)
+      coordinates[i, 1] = x[i, element]
+      coordinates[i, 2] = y[i, element]
+      if set_z_coordinate_zero == false
+        coordinates[i, 3] = u[i, element][variable_id]
+      end
+    end
+    trimesh[element] = GeometryBasics.normal_mesh(Makie.to_vertices(coordinates), makie_triangles)
+  end
+  plotting_mesh = merge([trimesh...]) # merge meshes on each element into one large mesh
+  return plotting_mesh
+end
+
+function generate_plotting_wireframe(pds::PlotDataSeries2D{<:UnstructuredPlotData2D};
+                                     set_z_coordinate_zero = false)
+  @unpack variable_id = pds
+  pd = pds.plot_data
+  @unpack xf, yf, uf = pd
+
+  if set_z_coordinate_zero==true
+    sol_f = zeros(eltype(first(uf)), size(xf)) # plot 2d surface by setting z coordinate to zero
+  else
+    sol_f = StructArrays.component(uf, variable_id)
+  end
+  xyz_wireframe = Makie.Point.(map(x->vec(vcat(x, fill(NaN, 1, size(x, 2)))), (xf, yf, sol_f))...)
+  return xyz_wireframe
+end
+
+"""
+    trixi_plot(sol::TrixiODESolution;
```

Comment:
I like `trixi_plot` since it's consistent with `trixi_include` or `trixi_test`. However, it seems to be a very generic name for a rather specific result (i.e., a very specific way of plotting), and also suggests that it somehow replaces a "standard" `plot` command (like it does for `trixi_include` and `trixi_test`). What about just using `plot_interactive` or `iplot` (in this case I'd argue that brevity might overrule descriptiveness)? Note that I do not have a fixed opinion or an optimal solution, it just feels slightly off as it is right now

Reply:
I like `iplot` - especially since I don't like the idea of having to type `plot_interactive` each time...

Reply:
It's only one character more (`iplot`: 5, `plot_` + TAB: 6). But I'm fine with `iplot` since that's what I do when calling this